fauna/client/client.py

```diff
@@ -1,19 +1,23 @@
 from datetime import timedelta
 from dataclasses import dataclass
 from typing import Any, Dict, Iterator, Mapping, Optional, List
+from contextlib import contextmanager
 
 import fauna
 from fauna.client.retryable import Retryable
 from fauna.errors import AuthenticationError, ClientError, ProtocolError, ServiceError, AuthorizationError, \
     ServiceInternalError, ServiceTimeoutError, ThrottlingError, QueryTimeoutError, QueryRuntimeError, \
-    QueryCheckError, ContendedTransactionError, AbortError, InvalidRequestError
+    QueryCheckError, ContendedTransactionError, AbortError, InvalidRequestError, RetryableFaunaException
 from fauna.client.headers import _DriverEnvironment, _Header, _Auth, Header
 from fauna.http.http_client import HTTPClient
 from fauna.query import Query, Page, fql
+from fauna.query.models import StreamToken
 from fauna.client.utils import _Environment, LastTxnTs
 from fauna.encoding import FaunaEncoder, FaunaDecoder
 from fauna.encoding import QuerySuccess, ConstraintFailure, QueryTags, QueryStats
 
+import httpx
```

Comment:
I'd like to avoid this. The goal was to encapsulate httpx inside the concrete httpx_client impl, which would allow users to bring their own http clients.

The current driver falls short of this (there's an import statement in an if/else branch when the global http client is none), but I don't want to move further away.

Can you think of a way to do this that keeps httpx outside of the client as a top level import?

Reply:
pretty sure I can do this